test/app_test.ts

```diff
@@ -1,57 +1,47 @@
-import { BlockchainLifecycle, web3Factory } from '@0x/dev-utils';
-import { runMigrationsOnceAsync } from '@0x/migrations';
-import { Web3ProviderEngine } from '@0x/subproviders';
-import { Web3Wrapper } from '@0x/web3-wrapper';
 import * as HttpStatus from 'http-status-codes';
-import 'mocha';
-import * as request from 'supertest';
 
-import { getAppAsync, getDefaultAppDependenciesAsync } from '../src/app';
-import * as config from '../src/config';
 import { DEFAULT_PAGE, DEFAULT_PER_PAGE, SRA_PATH } from '../src/constants';
 
-import { expect } from './utils/expect';
+import { defaultTestManager, TestCase } from './framework/test_manager';
 
-let app: Express.Application;
-
-let web3Wrapper: Web3Wrapper;
-let provider: Web3ProviderEngine;
-let accounts: string[];
-let blockchainLifecycle: BlockchainLifecycle;
-
-describe('app test', () => {
-    before(async () => {
-        // start ganache and run contract migrations
-        const ganacheConfigs = {
-            shouldUseInProcessGanache: false,
-            shouldAllowUnlimitedContractSize: true,
-            rpcUrl: config.ETHEREUM_RPC_URL,
-        };
-        provider = web3Factory.getRpcProvider(ganacheConfigs);
-        web3Wrapper = new Web3Wrapper(provider);
-        blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
-        await blockchainLifecycle.startAsync();
-        accounts = await web3Wrapper.getAvailableAddressesAsync();
-        const owner = accounts[0];
-        await runMigrationsOnceAsync(provider, { from: owner });
-
-        const dependencies = await getDefaultAppDependenciesAsync(provider, config);
-        // start the 0x-api app
-        app = await getAppAsync({ ...dependencies }, config);
-    });
-    it('should not be undefined', () => {
-        expect(app).to.not.be.undefined();
-    });
-    it('should respond to GET /sra/orders', async () => {
-        await request(app)
-            .get(`${SRA_PATH}/orders`)
-            .expect('Content-Type', /json/)
-            .expect(HttpStatus.OK)
-            .then(response => {
-                expect(response.body.perPage).to.equal(DEFAULT_PER_PAGE);
-                expect(response.body.page).to.equal(DEFAULT_PAGE);
-                expect(response.body.total).to.equal(0);
-                expect(response.body.records).to.deep.equal([]);
-            });
-    });
-});
+const manager = defaultTestManager();
+const suite: TestCase[] = [
+    {
+        description: 'should respond to GET /sra/orders',
+        action: {
+            actionType: 'apiGetRequestAsync',
```

Comment:
Could we make this an enum?

Reply:
We can, but that will really reduce the flexibility of the framework (this might be a good thing, but I'm not convinced of that right now). Something that might be a good middle ground is to create constants for common actions and assertions. I'll add this in the next commit.